model/bitmap-constructor.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+function Bitmap(buf) {
+  if(!buf) throw new Error('missing source buffer');
+  //tODO: Make buf a private property.
+  this.buf = buf;
+
+  if(this.getType() !== 'BM') throw new Error('unknown type, should be BM');
+
+  if(this.getNumColors() < 0) throw new Error('num colors should be a positive integer');
+
+  let w = this.getWidth();
+  let h = this.getHeight();
+  if(this.getPixelArray().length !== w * h) throw new Error('pixel array size does not equal w * h');
+
+  //note: the first row of the image is the bottom row
+}
+
+// ---------- GETTERS -----------------
+
+Bitmap.prototype.getType = function() {
+  return this.buf.toString('utf-8', 0, 2);
+};
+Bitmap.prototype.getWidth = function() {
+  return this.buf.readUInt32LE(18);
+};
+Bitmap.prototype.getHeight = function() {
+  return this.buf.readUInt32LE(22);
+};
+Bitmap.prototype.getNumColors = function() {
+  return this.buf.readInt32LE(46);
+};
+Bitmap.prototype.getColorArray = function() {
+  var numColors = this.buf.readInt32LE(46);
+  var dibSize = this.buf.readUInt32LE(14);
+
+  var colors = [];
+  var pos = dibSize + 14;
+  for(let i = 0; i < numColors; i++) {
+    var color = {};
+    color.blue  = this.buf.readUInt8(pos++);
+    color.green = this.buf.readUInt8(pos++);
+    color.red   = this.buf.readUInt8(pos++);
+    color.alpha = this.buf.readUInt8(pos++);
+    colors.push(color);
+  }
+  return colors;
+};
+Bitmap.prototype.getPixelArray = function() {
+  var pixelOffset = this.buf.readUInt32LE(10);
+  var numPixels = this.getWidth() * this.getHeight();
+  var pixels = [];
+  for (let i = 0; i < numPixels; i++) {
+    var offset = pixelOffset + i;
+    pixels.push(this.buf.readUInt8(offset));
+  }
+  return pixels;
+};
```

Comment:
@dbecker4130 @geoffsimons @meganreardon `getColorArray` and `getPixelArray` methods look good